cmake/FetchMongoC.cmake

```diff
@@ -0,0 +1,20 @@
+include (FetchContent)
+
+# Set the tag that we will fetch.
+set (MONGOC_FETCH_TAG_FOR_LIBBSON "1.17.0" CACHE STRING "The Git tag of mongo-c-driver that will be fetched to obtain libbson")
+
+# Fetch the source archive for the requested tag from GitHub
+FetchContent_Declare (
+    embedded_mcd
+    URL "https://github.com/mongodb/mongo-c-driver/archive/refs/tags/${MONGOC_FETCH_TAG_FOR_LIBBSON}.tar.gz"
```

Comment:
Poke. If deemed unnecessary/undesirable, feel free to reject this suggestion and resolve the conversation.

Reply:
Sorry I missed this. Indeed the `VERSION_CURRENT` logic is somewhat hairy and prevents this being workable without using git version tags. It would be nice to be able to fetch arbitrary branches and commits, though. One will likely want to do so eventually, but I'll kick that can down the road for now.